hazelcast/src/main/java/com/hazelcast/query/impl/IndexStore.java

```diff
@@ -127,9 +127,12 @@
     Set<QueryableEntry> evaluate(Predicate predicate, TypeConverter converter);
 
     /**
+     * @param descending whether the entries should come in the descending order.
+     *                   {@code true} means a descending order,
+     *                   {@code false} means an ascending order.
      * @return iterator over all index entries
      */
-    Iterator<QueryableEntry> getSqlRecordIterator();
+    Iterator<QueryableEntry> getSqlRecordIterator(boolean descending);
```

Comment:
As mentioned in the other comment, it seems strange that the `Iterator<QueryableEntry> getSqlRecordIterator(Comparable value)` overload doesn't take `descending` as an argument. Or is it because `EqualsFilter` always has values for all index components, and it is guaranteed that returned entries will have the same values for all columns of the desired collation?

Reply:
The `Iterator<QueryableEntry> getSqlRecordIterator(Comparable value)` is used only for all index key components and in this case, the ordering is not needed since all the values are equal.